develop/TOOLS/TOONuGet/NuGet_Naming_Conventions.md

```diff
@@ -0,0 +1,96 @@
+
+<br/>
+
+- Use the syntax: 'Skyline.DataMiner.\[Categories\].\[Name\]'. Optionally at the end you could have '.Protocol', '.Automation', '.Common', ... to specify on where it can be used (usage of SLManagedScripting, SLManagedAutomation, ...)
+    - Skyline.DataMiner.\[Categories\].\[Name\].Common (Shared code that can run either in Script or Protocol. Communicates with DataMiner via SLNet or [DataMinerSystem](https://www.nuget.org/packages?q=Skyline.DataMiner.Core.DataMinerSystem))
+    - Skyline.DataMiner.\[Categories\].\[Name\].Automation (Acts as entrypoint for Automation (methods for IEngine) or has a dependency on classes from SLManagedAutomation)
+    - Skyline.DataMiner.\[Categories\].\[Name\].Protocol (Acts as entrypoint for Protocol (methods for SLProtocol) or has a dependency on classes from SLManagedScripting)
+
+    - Another situation is the module that it has to do with. An example of this is the CI/CD packages:
+        - Skyline.DataMiner.CICD.DMApp.Common (common code for creating a package)
+        - Skyline.DataMiner.CICD.DMApp.Automation (specific code to convert an Automation solution to dmapp package)
+        - Skyline.DataMiner.CICD.DMApp.XXX (specific code to convert a XXX solution/repository to dmapp package)
+
+## Mandatory metadata
+
+```xml
+<PropertyGroup>
+    ...
+    <Authors>SkylineCommunications</Authors>
+    <Company>Skyline Communications</Company>
+    <Descrption>{Short description about the NuGet}</Description>
+    <PackageTags>Skyline;DataMiner</PackageTags>
+    <PackageProjectUrl>https://skyline.be</PackageProjectUrl>
+    <PackageReadmeFile>README.md</PackageReadmeFile>
+    ...
+</PropertyGroup>
+```
+
+- Authors has to be without whitespace.
+- PackageTags need to have at least 'Skyline' and 'DataMiner'. Extra tags can be added in wanted.
+- Always include a readme file to explain how to use the NuGet.
+
+When creating NuGets via the [SLC SE Repository Manager](xref:TOOSLCSERepositoryManager), the pipeline will automatically add the default license and icon file. If a different license or icon needs to be used then this can be specified in the metadata.
+
+```xml
+<PropertyGroup>
+    ...
+    <PackageLicenseFile>LICENSE.txt</PackageLicenseFile>
+    <PackageRequireLicenseAcceptance>True</PackageRequireLicenseAcceptance>
+    <PackageIcon>Icon.png</PackageIcon>
+    ...
+</PropertyGroup>
+```
+
+## Restricted names
+
+- Skyline.DataMiner.Files.XXX (DataMiner DLLs - Autogenerated by tool from Data-Acquisition domain)
+- Skyline.DataMiner.Dev.XXX (DevPacks - Autogenerated by tool from Data-Acquisition domain)
+- Skyline.DataMiner.CICD.XXX (Managed by Data-Acquisition domain)
+
+## Names to be used
+
+- Skyline.DataMiner.Core.XXX
+    - Holds the items that have a direct link to DataMiner (AppPackage creation/installation, DataMinerSystem, ...)
+    - DxM will also fall under this. Take in account the have clear error handling in case the DxM is optional and not available.
```

Comment:
Take in account the have clear error handling in case the DxM is optional and not available. => this remark feels a bit out of scope here.